test/framework/modules.py

```diff
@@ -56,6 +58,17 @@
 TEST_MODULES_COUNT = 81
 
 
+@contextmanager
+def temporary_home_dir():
+    tmpdir = tempfile.mkdtemp()
+    orig_home = os.environ['HOME']
+    os.environ['HOME'] = tmpdir
+    try:
+        yield tmpdir
+    finally:
+        os.environ['HOME'] = orig_home
+        shutil.rmtree(tmpdir)
```

Comment:
This is a bit much imho, just setting `os.environ['HOME'] = self.test_prefix` is the test is enough (`self.test_prefix` is cleaned up automatically in the tests, and the environment is restored before every test starts)

Reply:
Interesting...
However I did have failures due to test_prefix being used for other stuff too so a whole new ~world~ folder was needed. I'm also very much in favor for local judging of code and this is simple enough to understand (6 lines of code) and meaning in usage is very clear. So I'd keep this if you don't mind. Otherwise I'll change too `os.environ['HOME'] = tempfile.mkdtemp()` at least (IIRC tempfile is changed so the folder is a subfolder of test_prefix)

Reply:
I'd prefer the latter if you really need an empty dir for `$HOME` (less code to worry about, and you're just cleaning up what's already being cleaned up here)